### PR TITLE
Coalesce mutator

### PR DESCRIPTION
**Pull request by adeptofvoltron**
This PR:

- [x] Adds new feature ...
- [x] Covered by tests

### CODE REVIEW

**Comment by adeptofvoltron** [comment]
Last days travis is so slow...even on magento2 project travis do not work at all :/

File: src/Mutator/Operator/Coalesce.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    /**
+     * Replaces "'someValue' ?? 'otherValue';" with "(unset) 'someValue' ?? 'otherValue'"
+     *
+     * @param Node $node
+     *
+     * @return Node\Expr\BinaryOp\Coalesce
+     */
+    public function mutate(Node $node)
+    {
+        return new Node\Expr\BinaryOp\Coalesce(new Node\Expr\Cast\Unset_($node->left), $node->right);
```
Comment (adeptofvoltron) [inline src/Mutator/Operator/Coalesce.php]: it ought to be
`new Node\Expr\ConstFetch(new Node\Name('null'))`
instead 
`new Node\Expr\Cast\Unset_($node->left)`

Another thing to handle tommorow.

**Comment by borNfreee** [comment]
Travis reasonably fails with the following deprecation notice on PHP 7.2+:

```
Deprecated: The (unset) cast is deprecated
```

Why `(unset)` has been deprecated? Answer is here: https://wiki.php.net/rfc/deprecations_php_7_2

> The `(unset)` cast casts a value to `null`. This means that `(unset)` expr is simply an expression that always returns `null` and has no other side effects. Apart from being useless, this behavior is also confusing, as many people reasonably assume that `(unset) $a` will behave similarly to `unset($a)`, while in truth it does no such thing.

So, this is not a correct mutator. Moreover, it is the same as

```diff
- $a = $b ?? $c;
+ $a = null ?? $c;
```

which in its turn is the same as

```diff
- $a = $b ?? $c;
+ $a = $c;
```

File: src/Mutator/Operator/Coalesce.php (hunk 1 above)
Comment (sanmai) [reply src/Mutator/Operator/Coalesce.php]: Why not simply `$node->right`?

Comment (adeptofvoltron) [reply src/Mutator/Operator/Coalesce.php]: good point.

**Comment by sanmai** [comment]
It seems to me that AppVeyor build failures are unrelated.

**Comment by adeptofvoltron** [comment]
added. Still same troubles with AppVeyor

**Comment by adeptofvoltron** [comment]
added. Still some issues with AppVeyor...maybe they did something wrong during:
image

**Comment by borNfreee** [comment]
Thank you @adeptofvoltron

**Pull request merged**